test/gurka/gurka.cc

```diff
@@ -55,35 +55,40 @@ std::vector<midgard::PointLL> to_lls(const nodelayout& nodes,
 /**
  * build a valhalla json request body
  *
- * @param location_type  locations or shape
- * @param waypoints      sequence of pointlls representing the locations
+ * @param location_types vector of locations or shape, sources, targets
+ * @param waypoints      all pointll sequences for all location types
  * @param costing        which costing name to use, defaults to auto
  * @param options        overrides parts of the request, supports rapidjson pointer semantics
  * @param stop_type      break, through, via, break_through
  * @return json string
  */
-std::string build_valhalla_request(const std::string& location_type,
-                                   const std::vector<midgard::PointLL>& waypoints,
+std::string build_valhalla_request(const std::vector<std::string>& location_types,
+                                   const std::vector<std::vector<midgard::PointLL>>& waypoints,
```

Comment:
these are now vectors of their previous types as to not have to overload this internal function